tests/dummy/app/query-params/route.js

```diff
@@ -0,0 +1,21 @@
+import Ember from 'ember';
+import route from 'ember-redux/route';
+
+var queryParams = {
+  foo: {
+    refreshModel: false
+  }
+};
+
+function afterModel(dispatch, model, transition) {
+  if (Ember.isPresent(transition)) {
+    dispatch({
+      queryParams: Object.keys(this.get('queryParams')),
+      type: 'SET_QUERY_PARAMS'
+    });
+  }
+}
+
+var QueryParamsRoute = Ember.Route.extend();
```

Comment:
coming back to this now after reading the entire PR it seems we should maybe start on another route first (in the acceptance test) to assert the URL (assuming we can ensure no query params are present doing this?). Next we would transition to the query-params route (in which case the after model would fire) -asserting the diff after the dispatch fires. Finally what I feel is missing (maybe?) is that we should alter the query param in some way and see if that is correctly updated/reflected e2e. Could you maybe add a button to the child component and click it altering the queryParam in some way so we can see it's updated (executing the flow thru the reducer with a new value) ?

Maybe I just missed all that but from the test it seems we only flow thru 1x and I'd love to see us update from the default value if possible.

Ping me on slack today if I'm not clear :)

Also - thank you 100x for this pull request! Taking real world pain and patching it will save others pain down the road! thank you, thank you, thank you!